remote/remote_state.go

```diff
@@ -109,9 +109,18 @@ func (remoteState *RemoteState) differsFrom(existingBackend *TerraformBackend, t
 		}
 	}
 
-	if !reflect.DeepEqual(existingBackend.Config, remoteState.Config) {
-		terragruntOptions.Logger.Printf("Backend config has changed from %s to %s", existingBackend.Config, remoteState.Config)
-		return true
+	// Compare the states by looping through the map, assume it is different if either a key doesn't exist in the `existingBackend` or a value
+	// is different
+	for key, value := range remoteState.Config {
```

Comment:
This is only checking the top-level keys/values, whereas `DeepEqual` goes into any nested structures too, so this isn't quite doing the same check.